src/providers/tfc/workspaceProvider.ts

```diff
@@ -142,12 +158,46 @@ export class WorkspaceTreeItem extends vscode.TreeItem {
    * @param id This is the workspaceID as well as the unique ID for the treeitem
    * @param projectName The name of the project this workspace is in
    */
-  constructor(public name: string, public id: string, public projectName: string, public organization: string) {
+  constructor(
+    public name: string,
+    public id: string,
+    public projectName: string,
+    public organization: string,
+    public workspace: Workspace,
+    public weblink: vscode.Uri,
+    public lastRun?: RunAttributes,
+  ) {
```

Comment:
```suggestion
    public name: string,
    public projectName: string,
    public organization: string,
    public attributes: WorkspaceAttributes,
    public weblink: vscode.Uri,
    public lastRun?: RunAttributes,
  ) {
```
Related to the note above. Also replacing `workspace` with just the attributes allows us to call the tree item from the _outside_ as `workspace.attributes` or similar, more easily avoiding the awkward `workspace.workspace` which is something I ran into in https://github.com/hashicorp/vscode-terraform/pull/1475

btw. the doc comment above seems now outdated either way.